sources/api/ecs-settings-applier/build.rs

```diff
@@ -9,6 +9,17 @@ fn main() {
     println!("cargo:rerun-if-env-changed=VARIANT");
     if let Ok(variant) = env::var("VARIANT") {
         println!("cargo:rustc-cfg=variant=\"{}\"", variant);
+        let parts = variant.split('-').collect::<Vec<&str>>();
+        println!(
+            "cargo:rustc-cfg=variant_family=\"{}\"",
+            parts[0..2].join("-")
+        );
+        let variant_type = if parts.len() > 3 {
+            parts[3]
+        } else {
+            "general_purpose"
+        };
```

Comment:
Why not check for NVIDIA specifically? Is it because we'd also set `GPUSupportEnabled` for AMD and Inferentia?

Reply:
I didn't see support for AMD in the ECS agent, but to enable inferentia support, a different flag has to be (`InferentiaSupportEnabled`) set and we will set it based on the variant's name.

Reply:
It might be better to pin it to NVIDIA if `GPUSupportEnabled` is synonymous with that hardware.

Reply:
But we still need to know what "variant type" is built, since we could have `aws-k8s-*-inferentia` so that in `ecs-settings-applier` we do:

```rust
{
    GPUSupportEnabled : cfg!(variant_type = "nvidia"), 
    InferentiaSupportEnabled: cfg!(variant_type = "inferentia")
}
```